lib/game_new.rb

```diff
@@ -47,84 +48,82 @@ def game_begin
 
     loop do
       puts "Now, place your submarine!"
-      # puts @board.render(true)
       user_input = gets.chomp
       formatted = user_input.upcase.delete(",").split
-        if @board.valid_placement?(@submarine, formatted) 
-          @board.place(@submarine, formatted)
-          puts "Great placement!"
-          # puts @board.render(true)
-          
-          break
-        else
-          puts "Invalid coordinates, please try again."
-        end
-          # puts @board.render(true)
+      if @board.valid_placement?(@submarine, formatted) 
+        @board.place(@submarine, formatted)
+        puts "Great placement!"
+        break
+      else
+        puts "Invalid coordinates, please try again."
+      end
     end
   end
 
 
 
   def game_turn_start
-      
     loop do
       puts "=============COMPUTER BOARD============="
       puts @computer_board.render(true)
       puts "==============PLAYER BOARD=============="
       puts @board.render(true)
-        
-      # if @board.win == true
-      #   puts
-      # else
-      player_turn
-      computer_turn
+      if @computer_cruiser.health == 0 && @computer_submarine.health == 0
+        puts "you win!"
+        break
+      elsif @cruiser == 0 && @submarine == 0
+        puts "I win, you lose!"
+        break
+      else
+        player_turn
+        computer_turn
       end
+    end
   end
   
 
   def player_turn
     loop do
       puts "Enter the coordinate for your shot:"
       user_input = gets.chomp
-        formatted = user_input.upcase
-          if @computer_board.valid_coordinate?(formatted) && @computer_board.cells[formatted].fire_upon == true  #@computer_board.cells[formatted].fired_at == true 
-            if @computer_board.cells[formatted].empty? == false
-              if @computer_board.cells[formatted].render == "M"
-                puts "Your shot on #{formatted} was a miss."
-              elsif @computer_board.cells[formatted].render == "X"
-                puts "Your shot on #{formatted} sunk my ship!"
-              elsif @computer_board.cells[formatted].render == "H"
-                puts "Your shot on #{formatted} was a hit!"
-              end
-            end
-              
-              # puts @computer_board.render(true)
-            
-            break
-          else
-            puts "Please enter a valid coordinate:" 
+      formatted = user_input.upcase
+      if @computer_board.valid_coordinate?(formatted)   ##this is changing fired_at to true #@computer_board.cells[formatted].fired_at == true 
+        if @computer_board.cells[formatted].fired_upon? == false
```

Comment:
Good work on working out the fire_upon vs fired_upon? confusion. Looks good now!